src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -385,6 +384,17 @@ internal void SerializeEncoding(SerializationInfo info, StreamingContext context
             }
         }
 
+        // On Desktop, encoding instances that aren't cached in a static field are cached in
+        // a hash table by codepage.
+        private static volatile Hashtable encodings;
+#else
+        // On CoreCLR, the only instance that would need to be cached in the hash table is UTF32BE.
+        // Instead of using a hash table, simply cache the instance in a static field.
+        private static volatile Encoding s_utf32BE;
+        private static Encoding UTF32BE =>
```

Comment:
I was wondering the same thing, but wasn't sure if anyone would be opposed to the extra singleton. I'll go ahead and make that change.